### PR TITLE
[Enhancement] Support interpolation for Conditional Models + Modify BigGAN's forward function 

### PR DESCRIPTION
**Pull request by plyfager**
1. Modify BigGAN and BigGAN-Deep's forward function to support truncation, interpolation, and pre-trained weights with output channels order `RGB`.
2. Support interpolation for Conditional Models.

### CODE REVIEW

File: apps/interpolate_sample.py [hunk 1]
```diff
@@ -116,21 +148,38 @@ def sample_from_path(generator,
...
                      interp_mode='lerp',
                      space='z',
                      **kwargs):
-    interp_alphas = np.linspace(0, 1, intervals)
+    interp_alphas = torch.linspace(0, 1, intervals)
     interp_samples = []
 
+    if label_a is not None and label_b is not None:
+        device = get_module_device(generator)
+        embedding_a = generator.shared_embedding(label_a.to(device))
```
Comment (LeoXing1996) [inline apps/interpolate_sample.py:161]: May different models have different names for embedding layers?
Seems `shared_embedding` is only suitable for the BigGAN family?

Comment (plyfager) [reply apps/interpolate_sample.py:161]: True. I have two ideas, one is to set a unified name for all embedding layers, the other is to set a script argument for users to transfer the embedding layer's name. What's your opinion?

Comment (LeoXing1996) [reply apps/interpolate_sample.py:161]: Script argument may be better?
And we may also pre-define a `dict` for `model-embedding` name mapping?

e.g., `embedding_mapping = dict(BigGAN='shared_embedding')`

Then name of embedding can be accessed by

```[language=python]
if embedding_name is None:
    assert model_name in embedding_mapping, 'SOME INFO'
    embedding_name = embedding_mapping[model_name]

embedding_fn = getattr(generator, embedding_name)
embedding_a, embedding_b = embedding_fn(label_a.to(device)), embedding_fn(label_b.to(device))
```

Comment (nbei) [reply apps/interpolate_sample.py:161]: Some biggan models may not have `shared_embedding`? How do you deal with the models with a split embedding?

File: mmgen/models/architectures/biggan/generator_discriminator_deep.py [hunk 2]
```diff
@@ -269,7 +269,14 @@ def _get_default_arch_cfg(self, output_scale, base_channels):
...
 
-    def forward(self, noise, label=None, num_batches=0, return_noise=False):
+    def forward(self,
+                noise,
+                label=None,
+                num_batches=0,
+                return_noise=False,
+                truncation=-1.0,
+                use_embedding=True,
+                rgb2bgr=False):
```
[the remark this answers is not in the document]
Comment (plyfager) [reply mmgen/models/architectures/biggan/generator_discriminator_deep.py:279]: If this argument is defined in the initialization function, extra config files should be provided for users to use the converted weight?

**Comment by plyfager** [comment]
BigGAN models always have `shared_embedding` layer.  If a BIgGAN model degrades to an unconditional model, users should not 
use `--conditional`.

File: apps/interpolate_sample.py (hunk 1 above)
Comment (plyfager) [reply apps/interpolate_sample.py:161]: BigGAN models always have a `shared_embedding` layer. If a model has a split embedding,  `shared_embedding`  would be an `Identity` layer, and this case has also be tested.

File: mmgen/models/architectures/biggan/generator_discriminator.py [hunk 3]
```diff
@@ -256,7 +262,13 @@ def _get_default_arch_cfg(self, output_scale, base_channels):
...
         return _default_arch_cfgs[str(output_scale)]
 
-    def forward(self, noise, label=None, num_batches=0, return_noise=False):
+    def forward(self,
+                noise,
+                label=None,
+                num_batches=0,
+                return_noise=False,
+                truncation=-1.0,
+                use_embedding=True):
```
Comment (nbei) [inline mmgen/models/architectures/biggan/generator_discriminator.py:271]: `use_inner_embedding=True` may be better?

File: mmgen/models/architectures/biggan/generator_discriminator.py [hunk 4]
```diff
@@ -274,6 +286,13 @@ def forward(self, noise, label=None, num_batches=0, return_noise=False):
             return_noise (bool, optional): If True, ``noise_batch`` and
                 ``label`` will be returned in a dict with ``fake_img``.
                 Defaults to False.
+            truncation (float, optional): Truncation factor. Give value not
+                less than 0., the truncation trick will be adopted.
+                Otherwise, the truncation trick will not be adopted.
+                Defaults to -1..
+            use_embedding (bool, optional): Whether to use `shared_embedding`
```
Comment (nbei) [inline mmgen/models/architectures/biggan/generator_discriminator.py:293]: Is `shared_embedding`accurate? In Biggan, there may be some model with split embedding?

**Pull request merged**